lib/seattleflu/id3c/cli/command/etl/redcap_det_kiosk.py

```diff
@@ -44,11 +45,15 @@
     "kiosk",
     redcap_url = REDCAP_URL,
     project_id = PROJECT_ID,
-    required_instruments = REQUIRED_INSTRUMENTS,
     revision = REVISION,
     help = __doc__)
 
-def redcap_det_kisok(*, db: DatabaseSession, cache: TTLCache, det: dict, redcap_record: dict) -> Optional[dict]:
+@get_first_redcap_record_instance_only
+def redcap_det_kisok(*, db: DatabaseSession, cache: TTLCache, det: dict, redcap_record: REDCapRecord) -> Optional[dict]:
+    # Only process REDCap record if all required instruments are complete
+    if not required_instruments_complete(REQUIRED_INSTRUMENTS, redcap_record):
+        return None
```

Comment:
It seems natural to me to make this required instruments check a bit of declarative syntax via a decorator too. e.g. it moves from `@command_for_project(required_instruments = …)` to `@required_instruments(…)`.

The existing routines would then look like:

```python
@redcap_det.command_for_project(…)
@first_instance_of_record
@required_instruments(…)
def redcap_det_kiosk(…):
    …
```

Reply:
Hmm yeah that makes sense, I'll update.

Reply:
Updated in latest commit